src/Datagrid/Pager.php

```diff
@@ -838,10 +850,17 @@ public function getQuery()
     /**
      * @param int $nb
      *
+     * @deprecated since sonata-project/admin-bundle 3.86
+     *
      * @return void
      */
     protected function setNbResults($nb)
```

Comment:
This is used https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/src/Datagrid/Pager.php#L78
https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/blob/3.x/src/Datagrid/Pager.php#L53
so we still need a setter. 

Another way could be to stop implementing `countResult` in this class and let persistenceBundle doing it.

Reply:
Ah ok I see